src/landscape.cpp

```diff
@@ -1308,6 +1308,7 @@ static void BuildRiver(TileIndex begin, TileIndex end, TileIndex spring, bool ma
 	finder.FoundEndNode = River_FoundEndNode;
 	finder.user_target = &end;
 	finder.user_data = &user_data;
+	finder.max_search_nodes = _settings_game.pf.npf.npf_max_search_nodes;
```

Comment:
npf_max_search_nodes shouldn't ever be 0, so should it have been `_settings_newgame` instead of `_settings_game`?

While I agrree about configurability, slapping a 10000 in there seems wrong.